app/src/main/java/com/naxtlevelofandroiddevelopment/marvelgallery/presentation/common/Presenter.kt

```diff
@@ -0,0 +1,17 @@
+package com.naxtlevelofandroiddevelopment.marvelgallery.presentation.common
+
+import io.reactivex.disposables.Disposable
+
+abstract class Presenter {
+
+    var subscriptions: List<Disposable> = emptyList()
+
+    // Here view might not be loaded yet
+    open fun onCreate() {}
```

Comment:
Presenter without view is pretty useless, view can be passed to this base class and Presenter can be parametrised with view class type